src/Login/Login.jsx

```diff
@@ -96,6 +96,21 @@ class Login extends React.Component {
     }
     const customImageStyle = { backgroundImage: `url(/src/img/icons/${customImage}.svg)` };
 
+    const getLocationForText = (location) => {
+      let next = location.from;
+      if (location.state && location.state.from) {
+        next = location.state.from;
+      }
+      if (!next) {
+        return undefined;
+      }
+      // Lookup next to get actuale page name, if item is not in main navigation display default messaging
+      const nextItem = components.navigation.items.filter((item) => item.link === next)[0];
+
+      return nextItem ? nextItem.name : 'Restricted';
```

Comment:
This doesn't take care of homepage right? If logging in from homepage it will show as `Restricted` since `/` is not in `navigation.items`? And the message will just say `The page requires access`?

Reply:
I will look into, i was not aware that user can login from homepage. I think it should probably just show the default login page in this instance what are your thoughts? 

Do you know which environment has the homepage login? otherwise i will find it

Reply:
If you check on how `indexPublic` is being set in config that should work...
But you are right, it was implenent as a way of working with a very old Penegrine
Probably no body is using that any more and no need to worry

Reply:
I think taht is an edge case we can drop, I'll approve this

Reply:
I fixed, data.kidsfirstdrc.org still uses it and i made the change please review again